src/components/review/ReviewsList.js

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useState } from 'react';
+import axiosInstance from '../../apis/axiosClient';
+import WriteReview from './WriteReview';
+import { Alert } from '@mui/material';
+import { useAuth } from '@akosasante/react-auth-context';
+
+
+
+const AlbumReviews = ({ albumId }) => {
+  const [reviews, setReviews] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const [userHasReviewed, setUserHasReviewed] = useState(false);
+
+const { user } = useAuth(); //use user.user.<whatever field we want> to access it properly  
+
+  useEffect(() => {
+    fetchAlbumReviews();
+  }, [albumId]);
+
+  const fetchAlbumReviews = async () => {
+    try {
+      const response = await axiosInstance.get(`/reviews/album/${albumId}`);
+      console.log(response.data.allProductReviews);
+
+      const { allProductReviews } = response.data;
+      setReviews(allProductReviews);
+      // Check if the user has already reviewed the album
+      const hasReviewed = allProductReviews.some(review => review.user === user.user._id);
```

Comment:
This will cause errors if `user` is null (ie if a non-logged in user wants to take a look at reviews -- the reviews load fine, and the setReviews call happens sucessfully before this line, but then an error is caught here which causes the error message to be shown even though the reviews were loaded completely fine). We can put this in an `if` statement or use optional chaining on `user?.user?._id`

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/8ecacd0c-2dac-4a8c-a78c-6db09340e865)

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/bc2f3a00-2969-49ea-a237-e1758d65c6bd)

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/c1b77512-2d36-441d-9d27-ab381c874a03)

Reply:
@akosasante Thank you- with your changes everything works great! 
I am just wondering.. because
I've made sure I'm logged in and it throws an error (with no changes in code)- I just don't get why it sets user to _null_.

Reply:
Could you show the error it threw before? It could be that when the component first runs the `user` variable hasn't yet been loaded. React will re-render the component automatically when any of its props or variables its getting from hooks or `useState` change.